--- ngx_lua-0.10.21-orig/config 2022-03-02 14:54:22.000000000 +0800
+++ ngx_lua-0.10.21/config      2022-11-19 15:17:30.701888620 +0800
@@ -298,6 +298,7 @@
             $ngx_addon_dir/src/ngx_http_lua_input_filters.c \
             $ngx_addon_dir/src/ngx_http_lua_pipe.c \
             $ngx_addon_dir/src/ngx_http_lua_worker_thread.c \
+            $ngx_addon_dir/src/ngx_http_lua_ffi.c \
             "

 HTTP_LUA_DEPS=" \
