--- nginx-1.21.4-orig/src/core/ngx_thread_pool.c        2021-11-05 13:06:14.000000000 +0800
+++ nginx-1.21.4/src/core/ngx_thread_pool.c     2022-11-19 15:18:47.182642566 +0800
@@ -361,6 +361,26 @@
 }


+void
+ngx_thread_pool_notify_inject(void *data)
+{
+    ngx_thread_task_t *task = data;
+
+    task->next = NULL;
+
+    ngx_spinlock(&ngx_thread_pool_done_lock, 1, 2048);
+
+    *ngx_thread_pool_done.last = task;
+    ngx_thread_pool_done.last = &task->next;
+
+    ngx_memory_barrier();
+
+    ngx_unlock(&ngx_thread_pool_done_lock);
+
+    (void) ngx_notify(ngx_thread_pool_handler);
+}
+
+
 static void
 ngx_thread_pool_handler(ngx_event_t *ev)
 {
